src/compiler/scala/tools/nsc/backend/jvm/analysis/BackendUtils.scala

```
@@ -263,7 +263,10 @@ class BackendUtils[BT <: BTypes](val btypes: BT) {
 
     def visitInternalName(internalName: InternalName): Unit = if (internalName != null) {
       val t = classBTypeFromParsedClassfile(internalName)
-      if (t.isNestedClass.get) innerClasses += t
+      t.isNestedClass match {
+        case Right(true) => innerClasses += t
+        case _ => // ignores absent classes
+      }
```

I would argue that compilation should require this class to be on the classpath and fail otherwise. If we just ignore it here, the resulting classfile is different whether or not something is on the classpath or not, which is problematic IMO.

---

I see your point. OTOH, I'm hesitant to turn remaining bugs in our generic signatures into fatal errors as the code that generates them is somewhat brittle. Maybe a warning is appropriate?

---

I pushed a commit to make it fatal, we can run that through the community build to get some more data.

---

> turn remaining bugs in our generic signatures into fatal errors

My worry is that ignoring errors here applies not only to buggy generic signatures, but also ignores classfiles that are missing on the classpath. This should be an error I think.